src/app/selectors.js

```diff
@@ -169,6 +172,81 @@ const groupModelsByStatus = modelData => {
   return grouped;
 };
 
+/**
+  Returns a grouped collection of machine instances.
+  @param {Object} modelData
+  @returns {Function} The grouped machine instances.
+*/
+const groupMachinesByStatus = modelData => {
+  const grouped = {
+    blocked: [],
+    alert: [],
+    running: []
+  };
+  if (!modelData) {
+    return grouped;
+  }
+  for (let modelUUID in modelData) {
+    const model = modelData[modelUUID];
+    for (let machineID in model.machines) {
+      const machine = model.machines[machineID];
+      grouped[getMachineStatusGroup(machine).status].push(machine);
+    }
+  }
+  return grouped;
+};
+
+/**
+  Returns a grouped collection of unit instances.
+  @param {Object} modelData
+  @returns {Function} The grouped unit instances.
+*/
+const groupUnitsByStatus = modelData => {
+  const grouped = {
+    blocked: [],
+    alert: [],
+    running: []
+  };
+  if (!modelData) {
+    return grouped;
+  }
+  for (let modelUUID in modelData) {
+    const model = modelData[modelUUID];
+    for (let applicationID in model.applications) {
+      const application = model.applications[applicationID];
+      for (let unitID in application.units) {
+        const unit = application.units[unitID];
+        grouped[getUnitStatusGroup(unit).status].push(unit);
+      }
+    }
+  }
+  return grouped;
+};
+
+/**
+  Returns a grouped collection of machine instances.
+  @param {Object} modelData
+  @returns {Function} The grouped machine instances.
+*/
+const groupApplicationsByStatus = modelData => {
+  const grouped = {
+    blocked: [],
+    alert: [],
+    running: []
+  };
+  if (!modelData) {
+    return grouped;
+  }
+  for (let modelUUID in modelData) {
+    const model = modelData[modelUUID];
+    for (let applicationID in model.applications) {
+      const application = model.applications[applicationID];
+      grouped[getApplicationStatusGroup(application).status].push(application);
+    }
+  }
+  return grouped;
+};
+
```

Comment:
These three functions are so similar that I feel you should create a higher-order function for them where you could just pass `modelData` and the name of the collection?